pcdsdevices/slits.py

```diff
@@ -99,40 +100,36 @@ class Slits(Device, MvInterface):
-    the beam is being blocked or not depends on the pointing. In order to
-    create an estimate that will warn operators of narrowly closed slits while
-    still allowing slits to be closed along the beampath.
-
-    The simplest solution was to use a :attr:`.nominal_aperture` that stores
-    the slit width and height that the slits should use for general operation.
-    Using this the :attr:`.transmission` is calculated based on how the current
-    aperture compares to the nominal, always using the minimum of the width or
-    height. This means that if you have a nominal aperture of 2 mm, but your
-    slits are set to 0.5 mm, the total estimated transmission will be 25%.
-    Obviously this is greatly oversimplified, but it allows the lightpath to
-    make a rough back of the hand calculation without being over aggressive
-    about changing slit widths during alignment
+    The slits represent a unique device when forming the lightpath because
+    whether the beam is being blocked or not depends on the pointing. In order
+    to create an estimate that will warn operators of "closed" slits, we set a
+    ``nominal_aperture`` for each unique device along the beamline. This is
+    value is considered the smallest the slit aperture can become without
+    blocking the beamline. Both the ``width`` and the ``height`` need to exceed
+    this ``nominal_aperture`` for the ``Slits`` to be considered removed.
     """
-    xcenter = Cpt(SlitPositioner, '', slit_type="XCENTER", kind='hinted')
-    xwidth = Cpt(SlitPositioner, '', slit_type="XWIDTH", kind='normal')
-    ycenter = Cpt(SlitPositioner, '', slit_type="YCENTER", kind='hinted')
-    ywidth = Cpt(SlitPositioner, '', slit_type="YWIDTH", kind='normal')
+    xwidth = Cpt(SlitPositioner, '', slit_type="XWIDTH", kind='hinted')
+    ywidth = Cpt(SlitPositioner, '', slit_type="YWIDTH", kind='hinted')
+    nominal_aperture = Cpt(AttributeSignal, attr='_nominal', kind='normal')
+    xcenter = Cpt(SlitPositioner, '', slit_type="XCENTER", kind='normal')
+    ycenter = Cpt(SlitPositioner, '', slit_type="YCENTER", kind='normal')
     blocked = Cpt(EpicsSignalRO, ":BLOCKED", kind='omitted')
     open_cmd = Cpt(EpicsSignal, ":OPEN", kind='omitted')
     close_cmd = Cpt(EpicsSignal, ":CLOSE", kind='omitted')
     block_cmd = Cpt(EpicsSignal, ":BLOCK", kind='omitted')
-
     # Subscription information
     SUB_STATE = 'sub_state_changed'
     _default_sub = SUB_STATE
     # QIcon for UX
     _icon = 'fa.th-large'
 
-    def __init__(self, *args, nominal_aperture=(5.0, 5.0), **kwargs):
+    def __init__(self, *args, nominal_aperture=5.0, **kwargs):
         self._has_subscribed = False
-        self.nominal_aperture = nominal_aperture
+        self._nom = nominal_aperture
         super().__init__(*args, **kwargs)
+        # Modify Kind of center readbacks
+        self.xcenter.readback.kind = Kind.normal
+        self.ycenter.readback.kind = Kind.normal
```

Comment:
Are you required to import `Kind` to do these? Kind of annoying they don't just accept `'normal'`

Reply:
Haven't tried this.

Reply:
It worked